ciphers/rsa/RSAcipher.go

```diff
@@ -130,44 +135,63 @@ func toASCII(slice []rune) []int {
 	return converted
 }
 
-func toRune(slice []int) string {
+// ToRune convert a string to rune
+func ToRune(slice []int) string {
 	//runs in O(n) where n = len(slice)
-	var str string
-	for _, v := range slice {
-		str += string(v)
+	// var str string
+	var str strings.Builder
+	for i, v := range slice {
+		if i != len(slice)-1 {
+			str.WriteString(fmt.Sprintf("%d ", v))
+		} else {
+			str.WriteString(fmt.Sprint(v))
+		}
+
 	}
-	return str
+	return str.String()
 }
 
-func main() {
-	rand.Seed(time.Now().UTC().UnixNano())
-	bits := 17
-
-	p := generatePrimes(1 << bits)
-	q := generatePrimes(1 << bits)
-	for p == q {
-		q = generatePrimes(1 << bits)
+// Compare ... Added for manual comparison but didn't work
+func Compare(str string) []int {
+	a := strings.Split(str, " ")
+	fmt.Printf("%v", a)
+	var b []int
+	for _, v := range a {
+		temp, _ := strconv.Atoi(v)
+		b = append(b, temp)
 	}
+	return b
+}
 
-	n := p * q
+// func main() {
```

Comment:
Do let me know if you think that the main function that have been commented out should be removed completely.

I kept them here because I think they are useful for making new test cases, but a lot of them already have test files so I wasn't sure I should completely remove these blocks.

Reply:
Yes it should be removed completely.